### PR TITLE
Pipeline Launch: File Selection

### PR DESCRIPTION
**Pull request by joshsadam**
## Description of changes
Display a list of sample and their files **that can be run** on the currently selected pipeline.

image

## Related issue
Link to the GitHub issue this pull request addresses using the `#issuenum` format.  If it completes an issue, use `Fixes #issuenum` to automatically close the issue.

## Checklist
Things for the developer to confirm they've done before the PR should be accepted:

* [ ] CHANGELOG.md (and UPGRADING.md if necessary) updated with information for new change.
* [ ] Tests added (or description of how to test) for any new features.
* [ ] User documentation updated for UI or technical changes.

### CODE REVIEW

**Review by deepsidhu85 — COMMENTED** [review]
First thought looking at the image you attached. The email on pipeline completion checkbox is missing. Also, I don't know how I feel about the section headings with the line

**Comment by joshsadam** [comment]
What does the "email on pipeline completion" have to do with displaying files?  I can re-work the headings.

**Review by deepsidhu85 — CHANGES_REQUESTED** [review]
Overall looks good. Just a few things to change

File: src/main/webapp/resources/js/apis/pipelines/pipelines.js [hunk 1]
```diff
@@ -3,6 +3,7 @@
  */
 import axios from "axios";
 import { setBaseUrl } from "../../utilities/url-utilities";
+import { notification } from "antd";
```
[the remark this answers is not in the document]
Comment (joshsadam) [reply src/main/webapp/resources/js/apis/pipelines/pipelines.js:6]: Removed in c22841d88426c7aff7446e90af83e91ad7044b53

File: src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UIPipelineSampleService.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+	 * @param paired  Whether paired end files can be run on the current pipeline
+	 * @param singles Whether single end files can be run on the current pipeline
+	 * @return list of samples containing their associated sequencing data
+	 */
+	public List<LaunchSample> getPipelineSamples(boolean paired, boolean singles) {
+		Map<Project, List<Sample>> cart = cartService.getFullCart();
+		List<LaunchSample> samples = new ArrayList<>();
+		cart.forEach((project, projectSamples) -> {
+			for (Sample sample : projectSamples) {
+				LaunchSample ls = new LaunchSample(sample, project);
```
[the remark this answers is not in the document]
Comment (joshsadam) [reply src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UIPipelineSampleService.java:42]: Updated in 816dd5f6219aa976df98061c945d3d578c9dbac5

File: src/main/webapp/resources/js/pages/launch/LaunchFiles.jsx [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+
+  React.useEffect(() => {
+    fetchPipelineSamples({
+      paired,
+      singles,
+    })
+      .then((data) => {
+        setSamples(data);
+        setSelected(
+          data.filter((s) => s.files.length).map((s) => s.files[0].identifier)
```
[the remark this answers is not in the document]
Comment (joshsadam) [reply src/main/webapp/resources/js/pages/launch/LaunchFiles.jsx:44]: Updated in 6da8d9bff13a33be6bceec4b4f0a2a30a50ed075

File: src/main/webapp/resources/js/pages/launch/LaunchFiles.jsx [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+  }, [paired, singles]);
+
+  React.useEffect(() => {
+    setSelectedSampleFiles(dispatch, selected);
+  }, [selected]);
+
+  React.useEffect(() => {
+    if (samples) {
+      if (hideUnusable) {
+        setVisibleSamples(samples.filter((s) => s.files.length));
```
[the remark this answers is not in the document]
Comment (joshsadam) [reply src/main/webapp/resources/js/pages/launch/LaunchFiles.jsx:57]: Updated in ab4014ebb2261779550f82d5caffb6c5e7d95abf

File: src/main/webapp/resources/js/pages/launch/LaunchFiles.jsx [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+export function LaunchFiles() {
+  const [selected, setSelected] = React.useState();
+  const [
+    { acceptsPairedSequenceFiles: paired, acceptsSingleSequenceFiles: singles },
+    dispatch,
+  ] = useLaunch();
+  const [hideUnusable, setHideUnusable] = React.useState(true);
+  const [visibleSamples, setVisibleSamples] = React.useState();
+  const [samples, setSamples] = React.useState();
+
```
[the remark this answers is not in the document]
Comment (joshsadam) [reply src/main/webapp/resources/js/pages/launch/LaunchFiles.jsx:35]: Um... nope

File: src/main/webapp/resources/js/pages/launch/LaunchFiles.jsx [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+  const updateSelectedFiles = (previous, current) => {
+    const ids = new Set(selected);
+    ids.delete(previous);
+    ids.add(current);
+    setSelected(Array.from(ids));
+  };
+
+  const removeSampleFromCart = (sample, selectedId) => {
+    removeSample(sample.project.id, sample.id).then(() => {
+      setSamples(samples.filter((s) => sample.id !== s.id));
```
[the remark this answers is not in the document]
Comment (joshsadam) [reply src/main/webapp/resources/js/pages/launch/LaunchFiles.jsx:75]: I really like the letter `s`, this is the 3rd time you have asked me to change it, I am getting a little frustrated.

[the remark this answers is not in the document]
Comment (joshsadam) [reply src/main/webapp/resources/js/pages/launch/LaunchFiles.jsx:75]: I really like the letter `s`, this is the 3rd time you have asked me to change it, I am getting a little frustrated.

[the remark this answers is not in the document]
Comment (joshsadam) [reply src/main/webapp/resources/js/pages/launch/LaunchFiles.jsx:75]: Updated in 1f439228120aafd4fb6c7818526a305e0498fc0e

File: src/main/webapp/resources/js/pages/launch/LaunchFiles.jsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (joshsadam) [reply src/main/webapp/resources/js/pages/launch/LaunchFiles.jsx:35]: Ok, updated in 36217e186faf24e3afa542f73527c487c7048c32

**Review by deepsidhu85 — APPROVED** [review]

**Pull request merged**